### PR TITLE
Prepare for the move to "https://analytics.luminoso.com/api/v4/"

### PR DESCRIPTION
**Pull request by Tahnan**
Changes in this patch:
* Changing the URL_BASE as defined in constants.py
* Updating the URL in various docstrings (and a global in upload.py)
* Changing get_root_url to not assume that there is exactly one path component;
  rather, it now assumes that "v4" is part of the path.  (If and when the API
  changes to /api/v5/, we will have a new version of the client to go with it.)

### CODE REVIEW

**Comment by alin-luminoso** [comment]
Drive-by: this should definitely bump the version number, probably to 0.5.

File: luminoso_api/client.py [hunk 1]
```diff
@@ -489,12 +490,19 @@ def get_token_filename():
...
     """
     # make sure it's a complete URL, not a relative one
-    assert ':' in url
-    return '/'.join(url.split('/')[:4])
+    if '://' not in url:
+        raise ValueError('Please supply a full URL, beginning with http:// '
+                         'or https:// .')
+    if 'v4' not in url:
+        # We presumably have just, e.g., https://analytics.luminoso.com/ .
+        # Redirect properly.
```
Comment (alin-luminoso) [inline luminoso_api/client.py]: What if we have https://analytics.luminoso.com/api?

File: luminoso_api/constants.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
 from __future__ import unicode_literals
-API_VERSION=4
-API_HOST='api.luminoso.com'
-URL_BASE='https://%s/v%d' % (API_HOST, API_VERSION)
+API_VERSION = 4
```
Comment (alin-luminoso) [inline luminoso_api/constants.py]: Is this constant actually used anywhere on its own?  Else I'd really suggest just folding it into `URL_BASE`.

It is also somewhat contradictory that we have these constants here but don't use them in `get_root_url()`.  Maybe this should actually just export a hardcoded full `URL_BASE`?

File: luminoso_api/upload.py [hunk 3]
```diff
@@ -3,8 +3,8 @@
 from luminoso_api import LuminosoClient
 from luminoso_api.json_stream import transcode_to_stream, stream_json_lines
 
-ROOT_URL = 'https://api.luminoso.com/v4'
-LOCAL_URL = 'http://localhost:5000/v4'
+ROOT_URL = 'https://analytics.luminoso.com/api/v4/'
```
Comment (alin-luminoso) [inline luminoso_api/upload.py]: Is this not able to use the constant above?

**Comment by akenney** [comment]
The readme and changelog files should also get updated (and probably the client version number too).

File: README.md [hunk 4]
```diff
@@ -75,13 +75,13 @@ account_id under /projects:
...
 >>> print(projects)
-<LuminosoClient for https://api.luminoso.com/v4/projects/lumi-test/>
+<LuminosoClient for https://analytics.luminoso.com/api/v4/projects/lumi-test/>
 ```
 
 HTTP methods
 ------------
 
-The URLs you can communicate with are documented at https://api.luminoso.com/v4.
+The URLs you can communicate with are documented at https://analytics.luminoso.com/api/v4.
```
Comment (alin-luminoso) [inline README.md]: (drive-by: line length?)

Comment (Tahnan) [inline README.md]: line length? in a .md file?

Comment (alin-luminoso) [inline README.md]: ...looking it over apparently we just don't have a strong convention in this file.  Bleh.

File: luminoso_api/client.py [hunk 5]
```diff
@@ -487,14 +489,34 @@ def get_token_filename():
...
+        raise ValueError('Please supply a full URL, beginning with http:// '
+                         'or https:// .')
+
+    # If the path starts with v4, use that for the base URL, but issue a
+    # warning about the URL moving.
+    if parsed_url.path.startswith('/v4'):
+        if warn:
+            logger.warning('While the old URL will continue to work, please '
+                           'be aware that the Luminoso API has moved to %s',
+                           URL_BASE)
```
Comment (akenney) [inline luminoso_api/client.py]: This warning *only* applies to the deployed API (and maybe staging?), right?  Like, if I'm reusing lines from my ipython history and try to connect to "http://my-computer:5000/v4", it will print this warning and then try to use a URL that won't work?

Comment (Tahnan) [inline luminoso_api/client.py]: It applies to any API running via nginx (so it should also use a URL that will work if you connect to our master-staging API).

In general, though, you're right: the URL it gives will be the deployed API, and if you try to connect to your own computer at /v4, but your computer doesn't have a redirect set up (because it's not running the API through nginx), then yes, things will fail for you.  I am, however, fundamentally OK with this; I don't think it's necessary to go through any convoluted steps to make sure that *we* get the right error messages.  Warnings and redirects that only apply to external users seem fine to me.

**Comment by Tahnan** [comment]
Closing this and resubmitting the changes based on the "alpha" release instead.
